### PR TITLE
Add script to update gh-pages content

### PR DESCRIPTION
**Pull request by chalin**
Contributes to (semi-)automating #5

### CODE REVIEW

File: update-gh-pages.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+#
+# This script will build the API reference docs, copy them to the gh-pages
+# branch, and tell you whether you need to commit changes.
+
+set -e -o pipefail
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+: ${CURRENT_BRANCH:=master}
```
Comment (bshaffer) [inline update-gh-pages.sh]: I would rather the script fail if ran on a branch other than `master`. What is the use-case for publishing from a different branch?

File: update-gh-pages.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    echo "Switching back to branch $CURRENT_BRANCH"
+    (set -x; git checkout $CURRENT_BRANCH)
+  fi
+else
+  echo
+  echo "IMPORTANT:"
+  echo "  You're on the gh-pages branch now."
+  echo "  Pages of the API reference have changed; for details, run: git status"
+  echo "  You need to commit and push the changes to the upstream gh-pages branch."
+  exit 1
```
Comment (bshaffer) [inline update-gh-pages.sh]: So this script doesn't actually publish the changes, instead it exits with an error code if changes exist?

A better workflow would be it pushes the changes if they exist. That way we can run this in `.travis.yml` for the master branch only. Something like this:

```sh
else
    git add .
    git ci -am "Update reference docs from [SHA]"
    git push origin gh-pages
fi
```

File: update-gh-pages.sh (hunk 1 above)
Comment (chalin) [reply update-gh-pages.sh]: I couldn't have tested the script itself (on the current PR branch), if the restriction to `master` had been in place. I don't see the value in _preventing_ it from any other branch.

File: update-gh-pages.sh (hunk 2 above)
Comment (chalin) [reply update-gh-pages.sh]: As discussed, I've added a `--push` option.

**Comment by chalin** [comment]
@bshaffer and I agreed that we'll hold off on this PR.

**Pull request closed without merging**